app/main.py

```diff
@@ -2,24 +2,84 @@
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: type, name: str) -> None:
+        self.public_name = name
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: object, owner: type) -> int:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: object, value: int) -> None:
+        if self.min_amount <= value <= self.max_amount:
+            setattr(instance, self.protected_name, value)
+            if not hasattr(instance, "access"):
+                instance.access = True
+        else:
+            print(f"{instance.slide_name}: access is not granted"
+                  f" because of {self.public_name}({value}) "
+                  f"is not in range({self.min_amount}-{self.max_amount})")
```

Comment:
Define whether you place a space at the beginning or at the end of the string to improve your code style and be consistent:
```suggestion
            print(f"{instance.slide_name}: access is not granted "
                  f"because of {self.public_name}({value}) "
                  f"is not in range({self.min_amount}-{self.max_amount})")
```